src/main/java/ui/TestController.java

```diff
@@ -157,4 +159,30 @@ public static RepoIO createTestingRepoIO(Optional<JSONStore> jsonStoreToBeUsed)
         return new RepoIO(Optional.of(new DummySource()), jsonStoreToBeUsed,
                           Optional.of(RepoStore.TEST_DIRECTORY));
     }
+
+    /**
+     * Create an update progress window if not in test mode.
+     *
+     * @return update progress window
+     */
+    public static UpdateProgressWindow createUpdateProgressWindow() {
+        if (isTestMode()) {
+            return null;
```

Comment:
I think that's okay. Maybe there could be a wrapper in UI for that which the menu can call. The update manager already has the responsibility of managing the window, so it doesn't seem out of place. What do you think?